### PR TITLE
Fixed isCpuLimitEqual and isMemoryLimitEqual wrongly comparing null/undefined values.

### LINKED ISSUE #1730: [Bug]: Container size not recognized when CPU limit is not set
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

Downstream version (eg. `RHODS 1.29`)

### Version

RHODS 1.31

### Current Behavior

When I modify worspace sizes to ones that include both requests but limit only memory usage, after selecting they're applied correctly but later not recognized by ODS dashboard

### Expected Behavior

workspace sizes should be correctly recognized in dashboard

### Steps To Reproduce

1. Install RHODS/OpenDataHub
2. In dashborad config, modify noetbook sizes to include sizes that do not limit CPU
3. Create a new data science project
4. Add workspace with the new size
5. Notice that on the project dashboard new workspace is now visible with unknown size

### Workaround (if any)

Set CPU limits

### What browsers are you seeing the problem on?

Firefox, Chrome, Microsoft Edge

### Anything else

My reasoning behind not setting CPU limit comes from this blog post: https://home.robusta.dev/blog/stop-using-cpu-limits

### PR DESCRIPTION
**Pull request by tkolo**
Fixes #1730

## Description
Fixes an issue when notebook sizes contain a size with either CPU or memory limit not set. When viewing such noetbook in project details page, size was not recognized.

Before:
obraz

After:
obraz

## How Has This Been Tested?
1. Add noetbook size with no limits to ODH dashboard configuration
2. Create new DS project
3. Create workbench with new size
4. Go back to project details

Also added unit tests for good measure

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

@kywalker-rh @xianli123 (for dashboard change) and @Gkrumbach07 (for tests)

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by xianli123** [comment]
@tkolo Thanks for involving me. I will take a look at the original issue and this PR.

**Comment by xianli123** [comment]
Hi @tkolo I tried to reproduce the original issue #1730. However, I didn't find where I could set the CPU limits. I just saw all the options in the dropdown menu have the CPU limits. How can I get a container size without CPU limits? Thanks!
Screenshot 2023-09-05 at 16 51 19

**Comment by tkolo** [comment]
I wrote this in reproduction steps in original issue but forgot to include it here, sorry. You need to add your own custom size in dashboard config, as described here: https://ai-on-openshift.io/odh-rhods/configuration/#advanced-configuration

**Comment by xianli123** [comment]
That's a great doc. I found the ODC. But it seems I don't have the permission. May I need to request a new account to get the permission? 
Screenshot 2023-09-05 at 19 18 35

**Comment by tkolo** [comment]
Can't help really you here, I'm in role of `cluster-admin` in my sandbox environment so I have no such issues.

**Comment by xianli123** [comment]
Ok. I will try to get a sandbox environment. Thanks!

**Comment by tkolo** [comment]
I suggest using openshift local, per https://developers.redhat.com/products/openshift-local/overview
Podman has a plugin that provides GUI for installation process https://podman-desktop.io/docs/kubernetes/openshift/creating-an-openshift-local-cluster

**Comment by xianli123** [comment]
Hi @tkolo sorry for the late reply. I have checked the UI. I am curious about if it's easy for users to understand the "Limit test". What do you think about changing the wording from "Unknown" to "Not defined" or something like that?

**Comment by xianli123** [comment]
Hi @tkolo As for this issue, I agree with you. We can directly apply the values that are listed in the dropdown. So it looks good to merge this PR now. 

With regard to what you mentioned (quoted as below), we can separate a new issue, because we may need to discuss this part with others. What do you think?

> I'd channge questionmarks to a dash (-) or N/A or something like that, as questionmarks seem confusing there as well. 
image

**Comment by tkolo** [comment]
Sure, go ahead and create a new issue for that, feel free to tag me there. I can probably make a PR for whatever we discuss there as well

File: frontend/src/__tests__/unit/resourceEquality.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  expect(isMemoryLimitEqual('1Gi', '1Gi')).toBe(true);
+  expect(isMemoryLimitEqual('1Gi', '1024Mi')).toBe(true);
+  expect(isMemoryLimitEqual('1Gi', '1025Mi')).toBe(false);
+});
+
+test('Memory limit - correctly compares undefined values', () => {
+  expect(isMemoryLimitEqual('1Gi', undefined)).toBe(false);
+  expect(isMemoryLimitEqual('1024Mi', undefined)).toBe(false);
+  expect(isMemoryLimitEqual(undefined, undefined)).toBe(true);
+});
```
Comment (christianvogt) [inline frontend/src/__tests__/unit/resourceEquality.spec.ts:25]:
```suggestion
describe('isCpuLimitEqual', () => {
  test('correctly compares non-undefined values', () => {
    expect(isCpuLimitEqual('1', '1')).toBe(true);
    expect(isCpuLimitEqual('1000m', '1')).toBe(true);
    expect(isCpuLimitEqual('1001m', '1')).toBe(false);
  });

  test('correctly compares undefined values', () => {
    expect(isCpuLimitEqual('1000m', undefined)).toBe(false);
    expect(isCpuLimitEqual('1', undefined)).toBe(false);
    expect(isCpuLimitEqual(undefined, undefined)).toBe(true);
  });
});

describe('isMemoryLimitEqual', () => {
  test('correctly compares non-undefined values', () => {
    expect(isMemoryLimitEqual('1Gi', '1Gi')).toBe(true);
    expect(isMemoryLimitEqual('1Gi', '1024Mi')).toBe(true);
    expect(isMemoryLimitEqual('1Gi', '1025Mi')).toBe(false);
  });

  test('correctly compares undefined values', () => {
    expect(isMemoryLimitEqual('1Gi', undefined)).toBe(false);
    expect(isMemoryLimitEqual('1024Mi', undefined)).toBe(false);
    expect(isMemoryLimitEqual(undefined, undefined)).toBe(true);
  });
});
```

File: frontend/src/__tests__/unit/resourceEquality.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { isCpuLimitEqual, isMemoryLimitEqual } from '~/utilities/valueUnits';
```
Comment (christianvogt) [inline frontend/src/__tests__/unit/resourceEquality.spec.ts:1]: Thanks for adding unit tests!

We've recently moved to co-locating unit tests near their target source file. See https://github.com/opendatahub-io/odh-dashboard/pull/1773

Could you please rebase this PR and then relocate this test file to: `src/utilities/__tests__/valueUnits.spec.ts`

**Review by christianvogt — COMMENTED** [review]
Tested with a modified dashboard config and it shows up as shown in the screenshots.

File: frontend/src/__tests__/unit/resourceEquality.spec.ts (hunk 2 above)
Comment (tkolo) [reply frontend/src/__tests__/unit/resourceEquality.spec.ts:1]: sure

File: frontend/src/__tests__/unit/resourceEquality.spec.ts (hunk 1 above)
Comment (christianvogt) [reply frontend/src/__tests__/unit/resourceEquality.spec.ts]: @tkolo Could you make the suggested change to group the tests.

Comment (tkolo) [reply frontend/src/__tests__/unit/resourceEquality.spec.ts]: Ops, forgot about that part, sorry about that. Fixed

**Review by xianli123 — APPROVED** [review]

**Pull request merged**